src/main/java/org/onebusaway/alexa/handlers/intent/YesIntentHandler.java

```diff
@@ -0,0 +1,91 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.onebusaway.alexa.handlers.intent;
+
+import com.amazon.ask.model.Response;
+import lombok.extern.log4j.Log4j;
+import org.apache.commons.lang3.StringUtils;
+import org.onebusaway.alexa.constant.Prompt;
+import org.onebusaway.alexa.constant.SessionAttribute;
+import org.onebusaway.alexa.exception.OneBusAwayException;
+import org.onebusaway.alexa.lib.ObaUserClient;
+import org.onebusaway.alexa.storage.ObaUserDataItem;
+import org.onebusaway.alexa.util.CityUtil;
+import org.onebusaway.alexa.util.RouteFilterUtil;
+import org.onebusaway.alexa.util.StopUtil;
+
+import java.util.Optional;
+
+import static org.onebusaway.alexa.constant.SessionAttribute.ASK_STATE;
+
+/**
+ * Handler for Yes intent request(e.g. user say "Yes" during the dialog).
+ */
+@Log4j
+public class YesIntentHandler extends IntentHandler {
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getIntentRequestName() {
+        return "AMAZON.YesIntent";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Optional<Response> handleWithObaData(final ObaUserDataItem obaUserDataItem, final ObaUserClient obaUserClient) {
+        if (askState == SessionAttribute.AskState.VERIFYSTOP) {
+            addOrUpdateSessionAttribute(ASK_STATE, askState);
+            return handleWithoutObaData();
+        }
+
+        if (askState == SessionAttribute.AskState.FILTER_INDIVIDUAL_ROUTE) {
+            return RouteFilterUtil.handleFilterRouteResponse(this.attributesManager, true, obaDao, obaUserDataItem);
+        }
+
+        log.error("Received yes intent without a question.");
+        return promptHelper.getResponse(Prompt.GENERAL_ERROR_MESSAGE);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Optional<Response> handleWithoutObaData() {
+        if (askState == SessionAttribute.AskState.VERIFYSTOP) {
+            return StopUtil.handleDuplicateStopResponse(personalization.getPrincipleId(), attributesManager, true, googleMaps, obaClient, obaDao);
+        }
+
+        if (askState == SessionAttribute.AskState.COPY_PROFILE_CONFIRM && personalization.isPersonalized()) {
+            try {
+                Optional<ObaUserDataItem> obaUserDataItem = this.obaDao.getUserData(personalization.getUserId());
+                obaUserDataItem.get().setUserId(personalization.getPersonId());
+                obaUserDataItem.get().setPreviousResponse(StringUtils.EMPTY);
+                ObaUserClient obaUserClient = new ObaUserClient(obaUserDataItem.get().getObaBaseUrl());
+                this.obaDao.saveUserData(obaUserDataItem.get());
+                return CityUtil.tellArrivals(obaUserDataItem.get(), obaUserClient, attributesManager, obaDao);
```

Comment:
When copying the profile, the route filters aren't used when reading the arrivals to the user immediately after copying the profile settings - maybe the session attributes aren't loaded here?

Note that when launching the skill again after onboarding with personalization, route filters are correctly recognized.

Similarly, we need to make sure the other profile settings like clock times are also used when reading the arrivals immediately after the copy.

#### Steps to repro

1. Create a profile without personalization
2. Set a route filter on a stop
3. Turn on personalization and launch the skill
4. Say "Yes" to copy the profile info

#### What I hear

Alexa reads all the arrivals for the stop, ignoring the route filter

#### What I expect

Alexa should honor the route filter and only read arrivals for routes for which I haven't filtered out arrivals

Reply:
Good callout, I will populate all attributes to session when we copy the profile